app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,48 @@
+class AddressesController < ApplicationController
+  def index 
+    @addresses = Current.user.addresses
+  end
+
+  def new
+    @address = Address.new
+  end
+
+  def create
+    @address = Current.user.addresses.new(address_params)
+    if @address.save
+      flash[:success] = "Address Added successfully"
+      redirect_to user_addresses_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @address = Address.find(params[:id])
```

Comment:
@03mayank, this should be `Current.user.addresses.find(params[:id])` right? Otherwise, a user will be able to update another user's address.